### DISCUSSION TITLE
RandomLinkSplit for direct HeteroData

### DISCUSSION THREAD

**Discussion by siam1991** [pyg-team/pytorch_geometric#5219] [Q&A]
hi:
I have a direct HeteroData，and I want to use RandomLinkSplit to split this dataset，but I don't know the "rev_edge_types" param should be?
I read the source code, find the length of edge_types should equal to rev_edge_types
if the edge_types is 
```
[('paper', 'cites', 'paper'), ('author', 'writes', 'paper'), ('author', 'affiliated_with', 'institution')],
```
what should the rev_edge_types is?

**Comment by Padarn** [comment]
The reverse edges are ones that you want to be thought of as the equivalent of the forward edge so that they are sampled together. If you don't have reverse edges you can ignore this, if you do have reverse edges you should include this.

If some have reverse and others don't you can provide none, so in your case you could have 
```
edge_types = [('paper', 'cites', 'paper'), ('author', 'affiliated_with', 'institution')]
rev_edge_types = [('author', 'writes', 'paper'), None]
```

**Reply by rusty1s** [reply]
Thanks for clarifying @Padarn. To add, defining reverse edges will be important to prevent data leakage within your split. That is, we need to ensure that when we split based on edges, the reverse edges will land in the same split.
